test/clj/rems/application/test_eraser.clj

```diff
@@ -1,58 +1,235 @@
+                  (let [events (expiration-notification-events draft-expires-in-6d)]
+                    (is (= 1 (count events)))
+                    (is (time/equal? (time/plus now (time/days 6)) (:expires-on (first events))))
+                    (is (time/equal? (time/minus now (time/days 84)) (:last-activity (first events))))
+                    (is (time/equal? now (:event-time (first events))))))
+                (is (= [{:outbox/deadline (time/date-time 2022)
+                         :outbox/email {:subject (str "Your unsubmitted application " draft-expires-in-6d " will be deleted soon")
+                                        :body (str "Dear alice,\n\n"
+                                                   "Your unsubmitted application has been inactive since 2021-10-09 and it will be deleted after 2022-01-07, if it is not edited.\n\n"
+                                                   "You can view and edit the application at localhost/application/" draft-expires-in-6d)
+                                        :to-user "alice"}
+                         :outbox/type :email}
+                        {:outbox/deadline (time/date-time 2022)
+                         :outbox/email {:subject (str "Your unsubmitted application " draft-expires-in-6d " will be deleted soon")
+                                        :body (str "Dear member,\n\n"
+                                                   "Your unsubmitted application has been inactive since 2021-10-09 and it will be deleted after 2022-01-07, if it is not edited.\n\n"
+                                                   "You can view and edit the application at localhost/application/" draft-expires-in-6d)
+                                        :to-user "member"}
+                         :outbox/type :email}]
+                       @outbox-emails))
+
+                (testing "processing applications again should not send new notifications"
+                  (eraser/process-applications!)
+                  (is (= #{draft-expires-in-6d draft-expires-in-8d submitted}
+                         (set (get-all-application-ids "alice")))
+                      "no applications have been deleted")
+                  (is (empty? (expiration-notification-events draft-expires-in-8d)))
+                  (is (empty? (expiration-notification-events submitted)))
+
+                  (testing "expiration notifications events have not changed for draft expiring in 6 days"
+                    (let [events (expiration-notification-events draft-expires-in-6d)]
+                      (is (= 1 (count events)))
+                      (is (time/equal? (time/plus now (time/days 6)) (:expires-on (first events))))
+                      (is (time/equal? (time/minus now (time/days 84)) (:last-activity (first events))))
+                      (is (time/equal? now (:event-time (first events))))))
+                  (is (= [{:outbox/deadline (time/date-time 2022)
```

Comment:
And then perhaps we can skip testing for this one (it's a lot of duplicate code).